hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/ECReplicationConfig.java

```diff
@@ -36,6 +43,29 @@ public ECReplicationConfig(int data, int parity) {
     this.parity = parity;
   }
 
+  public ECReplicationConfig(String string) {
+    final Pattern pattern = Pattern.compile("(\\d+)-(\\d+)");
+    final Matcher matcher = pattern.matcher(string);
+    if (!matcher.matches()) {
+      throw new IllegalArgumentException("EC replication config should be " +
+          "defined in the form 3-2, 6-3 or 10-4");
+    }
+
+    data = Integer.parseInt(matcher.group(1));
+    parity = Integer.parseInt(matcher.group(2));
+    if (data <= 0 || parity <= 0) {
+      throw new IllegalArgumentException("Data and parity part in EC " +
+          "replication config supposed to be positive numbers");
+    }
+    if ((data == 3 && parity == 2)
+        || (data == 6 && parity == 3)
+        || (data == 10 && parity == 4)) {
+      return;
+    }
+    LOG.warn("We recommend to use EC 3-2, 6-3 or 10-4. Other combinations " +
```

Comment:
Should we throw an exception here?

Reply:
I prefer to restrict to allow only supported combinations. So, +1 to throw exception here.